tests/tas-elasticsearch/src/test/java/org/alfresco/elasticsearch/reindexing/PathUpdateTests.java

```diff
@@ -0,0 +1,462 @@
+
+import org.alfresco.elasticsearch.SearchQueryService;
+import org.alfresco.rest.core.RestWrapper;
+import org.alfresco.rest.model.RestCategoryLinkBodyModel;
+import org.alfresco.rest.model.RestCategoryModel;
+import org.alfresco.rest.model.RestNodeBodyMoveCopyModel;
+import org.alfresco.rest.model.RestNodeModel;
+import org.alfresco.rest.search.SearchRequest;
+import org.alfresco.tas.AlfrescoStackInitializer;
+import org.alfresco.utility.data.DataContent;
+import org.alfresco.utility.data.DataSite;
+import org.alfresco.utility.data.DataUser;
+import org.alfresco.utility.data.RandomData;
+import org.alfresco.utility.model.ContentModel;
+import org.alfresco.utility.model.FileModel;
+import org.alfresco.utility.model.FolderModel;
+import org.alfresco.utility.model.SiteModel;
+import org.alfresco.utility.model.UserModel;
+import org.alfresco.utility.network.ServerHealth;
+import org.alfresco.utility.report.log.Step;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.testng.AbstractTestNGSpringContextTests;
+import org.testcontainers.shaded.org.apache.commons.lang3.tuple.Triple;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+/**
+ * Tests to check that paths are updated correctly.
+ * <p>
+ * These tests use an unusual structure where the bulk of the code is stored in lambdas within the {@#testSteps} method.
+ * All the "given" and "when" steps are run in the `@BeforeClass` method (with a one second pause between them) and then
+ * the "then" steps are run in the individual `@Test` methods.
+ * This structure saves about two seconds per test over having the "given" and "when" steps in the `@Test` methods.
+ */
+@ContextConfiguration (locations = "classpath:alfresco-elasticsearch-context.xml",
+        initializers = AlfrescoStackInitializer.class)
+public class PathUpdateTests extends AbstractTestNGSpringContextTests
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(PathUpdateTests.class);
+
+    @Autowired
+    private ServerHealth serverHealth;
+    @Autowired
+    private DataUser dataUser;
+    @Autowired
+    private DataSite dataSite;
+    @Autowired
+    private DataContent dataContent;
+    @Autowired
+    protected RestWrapper restClient;
+    @Autowired
+    protected SearchQueryService searchQueryService;
+
+    private UserModel testUser;
+    private SiteModel testSite;
+
+    /** Map from test method name to the set of named properties used by the test. */
+    private Map<String, Map<String, Object>> propertyMaps;
+
+    /** All the "given", "when" and "then" steps for each test, grouped according to the test name. */
+    private Map<String, Triple<Supplier<Map<String, Object>>, Function<Map<String, Object>, Map<String, Object>>, Consumer<Map<String, Object>>>> testSteps = Map.of(
```

Comment:
Maybe it's worth considering hiding this crazy definition in some class, like `TestSteps extends Map<...>`. This could allow also to hide the logic of tests methods verification, particular steps execution, etc.

Reply:
Yeah - that sounds good. Potentially we could have any number of step initialisation steps separated by waits, followed by a verification step for each test.